components/dashboard/src/user-settings/Account.tsx

```diff
@@ -30,18 +30,9 @@ export default function Account() {
         if (!user) {
             return;
         }
-        if (User.isOrganizationOwned(user)) {
-            const compareTime = (a?: string, b?: string) => (a || "").localeCompare(b || "");
-            const recentlyUsedSSOIdentity = user.identities
-                .sort((a, b) => compareTime(a.lastSigninTime, b.lastSigninTime))
-                // optimistically pick the most recent one
-                .reverse()[0];
-            setEmail(recentlyUsedSSOIdentity?.primaryEmail!);
-        } else {
-            const primaryEmail = User.getPrimaryEmail(user!);
-            if (primaryEmail) {
-                setEmail(primaryEmail);
-            }
+        const primaryEmail = User.getPrimaryEmail(user!);
```

Comment:
I think we can remove a bunch of this now that getPrimaryEmail() handles the logic, with it being called in `User.getProfile()` already, we have the state defaulting we need. I can push a commit to this branch that adjusts that, and addresses a bug where you can't edit your email even when you should be able to.